docs/source/ContributionGuide.rst

```diff
@@ -65,10 +65,119 @@ be submitted as a github Pull Request. This process tries to ensure that every
 contributed patch goes through the CI jobs before it is considered good for
 inclusion.
 
+Git workflow
+------------
+
+- Fork the repository in github.
+
+- Clone from your fork::
+
+    $ git clone git@github.com:<username>/avocado.git
+
+- Enter the directory::
+
+    $ cd avocado
+
+- Create a ``remote``, pointing to the upstream::
+
+    $ git remote add upstream git@github.com:avocado-framework/avocado-misc-tests.git
+
+- Configure your name and e-mail in git::
+
+    $ git config --global user.name "Your Name"
+    $ git config --global user.email email@foo.bar
+
+- Golden tip: never work on local branch master. Instead, create a new
+  local branch and checkout to it::
+
+    $ git checkout -b my_new_local_branch
+
+- Code and then commit your changes::
+
+    $ git add new-file.py
+    $ git commit -a -s
```

Comment:
I'd skip the `-a` as you already added the changes by `git add`. (alternatively I'd mention use:

```
$ git add new-file.py
$ git commit -s
# or "git commit -as" to commit all changes
```